text/0000-experimental_keywords.md

```diff
@@ -0,0 +1,84 @@
+
+Most experimental keywords are given a final syntax intended for general, long-term use, some time after the feature itself becomes available in the Stable compiler. This way the Rust community has an opportunity to try it out and see how it feels, how they use it in practice, and so on. Some experimental keywords aren't ever given a direct syntax of their own, instead they're simply there so that proc-macros can expand to code that uses the keyword.
+
+[And here we'd ideally we'd add at least one or two real examples to the guide once we have some examples of this in practice]
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+This RFC alters the language so that `r # $ token` is an accepted token sequence.
+* It names an experimental keyword named `token`.
+* The experimental keyword named _must_ be known to the compiler. If you attempt to name an experimental keyword that the compiler doesn't recognize it is an error.
+* The exact meaning and usage for each experimental keyword depends on the language feature it's associated with, and is not specified here.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* It can slightly hurt Rust's stability story to deliberately introduce new keywords that are intended to become deprecated within a relatively short period of time.
+  * Mitigation 1: Experimental keywords that do become Stable would continue to work even once the final syntax is decided upon and also stabilized. Stable code would not break.
+  * Mitigation 2: The rather unusual `r#$` prefix stands out as "clearly unusual"
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+* Alternative: We could select _some other_ sequence of invalid tokens to use as a prefix for experimental keywords.
+  * `r#$` is used because it is close enough to the "raw literal" and "raw string" syntaxes that people are very likely to understand what's going on even if they're not familiar with a specific experimental keyword when they first see it.
+
+* Alternative: We could limit the experimental keyword usage to Nightly only until the final syntax is decided upon.
+  * This would save people who only use the Stable channel from having to worry about using a new feature one way and then potentially being encouraged to update to a second syntax later.
+  * However, in the 2019 Survey, only 30% of users responded that they use Nightly, so a very large portion of the community would end up excluded from the experimentation phase.
+
+# Prior art
+[prior-art]: #prior-art
+
+* The Ember framework for javascript places experimental framework abilities into a special module that is clearly experimental so that users can try new things and become familiar with new features even before the feature is fully stable.
```

Comment:
```suggestion
* The Ember framework for JavaScript places experimental framework abilities into a special module that is clearly experimental so that users can try new things and become familiar with new features even before the feature is fully stable.
```